types/rlp_marshal.go

```diff
@@ -194,5 +194,9 @@ func (t *Transaction) MarshalRLPWith(arena *fastrlp.Arena) *fastrlp.Value {
 		vv.Set(arena.NewBytes([]byte{byte(t.Type)}))
 	}
 
+	if t.IsStateTx() {
```

Comment:
Maybe we should serialize `from` only if it is not equal to `contracts.SystemCaller` and in that way save `20` bytes per state transaction

Reply:
In that case I would need to consider use case when `From` field is missing like it has `contracts.SystemCaller` value assigned, which is not entirely correct (unable to make difference between it and assigning `nil` value).

Btw I need `From` field serialized, because of this check:
https://github.com/0xPolygon/polygon-edge/pull/796/files#diff-f64dae91a36980ba02c538f69fb83b7c80a8b4345db96849127307d33ccfacd1R882

Reply:
I made a Jira task for this, so we can come back to that optimization later: https://polygon.atlassian.net/browse/EVM-146